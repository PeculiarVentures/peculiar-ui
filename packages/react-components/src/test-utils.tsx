import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
