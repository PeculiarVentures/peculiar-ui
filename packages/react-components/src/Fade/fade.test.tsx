import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Fade } from './index';

describe('<Fade />', () => {
  it('should render with default styles', () => {
    const {
      asFragment,
    } = render(
      <Fade>
        <div>Inside</div>
      </Fade>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should finish animation', () => {
    const {
      asFragment,
    } = render(
      <Fade in>
        <div>Inside</div>
      </Fade>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
