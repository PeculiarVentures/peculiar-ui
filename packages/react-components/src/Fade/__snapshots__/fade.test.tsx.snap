// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Fade /> > should finish animation 1`] = `
<DocumentFragment>
  <div
    style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    Inside
  </div>
</DocumentFragment>
`;

exports[`<Fade /> > should render with default styles 1`] = `
<DocumentFragment>
  <div
    style="opacity: 0; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; visibility: hidden;"
  >
    Inside
  </div>
</DocumentFragment>
`;
